controller/destination/k8s_resolver_test.go

```diff
@@ -124,13 +115,9 @@ func TestK8sResolver(t *testing.T) {
 		cancelFn()
 		<-done
 
-		if mockDnsWatcher.ListenerSubscribed != listener || mockDnsWatcher.ListenerUnsubscribed != listener {
+		if mockEndpointsWatcher.ListenerSubscribed != listener || mockEndpointsWatcher.ListenerUnsubscribed != listener {
 			t.Fatalf("Expected listener [%v] to have been subscribed then unsubscribed to endpoint watcher, got: %+v", listener, mockEndpointsWatcher)
 		}
-
-		if mockEndpointsWatcher.ListenerSubscribed != nil || mockEndpointsWatcher.ListenerUnsubscribed != nil {
-			t.Fatalf("Did not expect listener [%v] to have been subscribed then unsubscribed to dns watcher, got: %+v", listener, mockEndpointsWatcher)
-		}
 	})
 
 	//TODO: Resolve name using DNS similar to Kubernetes' ClusterFirst
```

Comment:
Would you mind removing this TODO as well as part of this branch?

Reply:
> Would you mind removing this TODO as well as part of this branch?

I am going to remove it, and other similar things, in a separate PR.

Reply:
You're removing the original comment from which this comment was copied (line 51, k8s_resolver.go) as part of this branch. I think it makes sense to also remove this comment simultaneously.